crates/rpc/src/server/accept.rs

```diff
@@ -75,148 +182,232 @@ where
+            params,
+            quality,
+        })
+    }
 
-    fn try_from(input: &'a str) -> Result<Self, Self::Error> {
-        let (input, _) = tag("application/vnd.")(input)?;
-        let (input, app_name) = take_until(".")(input)?;
-        let (input, _) = tag(".")(input)?;
-        let (input, version) = take_until("+")(input)?;
-        let (response_type, _) = tag("+")(input)?;
+    fn is_miden_application(&self) -> bool {
+        // The subtype checks prefix only to include the legacy `vnd.miden+grpc` and optional
+        // version.
+        self.main.matches("application")
+            && (self.subtype.matches("vnd.miden") || self.subtype.starts_with("vnd.miden+grpc"))
+    }
 
-        Ok(AcceptHeaderValue { app_name, version, response_type })
+    fn is_obsolete(&self) -> bool {
+        self.subtype.starts_with("vnd.miden+grpc")
     }
 }
 
-// ACCEPT TESTS
+// HEADER VERIFICATION TESTS
 // ================================================================================================
 
 #[cfg(test)]
 mod tests {
+    use http::{HeaderMap, HeaderValue, header::ACCEPT};
+    use miden_objects::Word;
     use semver::Version;
 
-    use super::{AcceptHeaderValue, AcceptLayer};
+    use super::AcceptHeaderLayer;
 
-    #[test]
-    fn current_version_is_parsed_and_matches() {
-        let a = AcceptLayer::new().unwrap();
-        let version = env!("CARGO_PKG_VERSION");
-        let version = Version::parse(version).unwrap();
-        assert!(a.version_req.matches(&version));
-    }
+    const TEST_GENESIS_COMMITMENT: &str =
+        "0x00000000000000000000000000000000000000000000000000000000deadbeef";
+    const TEST_RPC_VERSION: Version = Version::new(1, 2, 3);
 
-    #[test]
-    fn same_minor_different_patch_matches() {
-        let a = AcceptLayer::new().unwrap();
-        let version = env!("CARGO_PKG_VERSION");
-        let mut version = Version::parse(version).unwrap();
-        version.patch += 1;
-        assert!(a.version_req.matches(&version));
+    impl AcceptHeaderLayer {
+        fn for_tests() -> Self {
+            Self::new(TEST_RPC_VERSION, Word::try_from(TEST_GENESIS_COMMITMENT).unwrap())
+        }
     }
 
+    #[rstest::rstest]
+    #[case::empty("")]
+    #[case::wildcard("*/*")]
+    #[case::wildcard_subtype("application/*")]
+    #[case::media_type_only("application/vnd.miden")]
+    #[case::with_quality("application/vnd.miden; q=0.3")]
+    #[case::version_exact("application/vnd.miden; version==1.2.3")]
+    #[case::version_major("application/vnd.miden; version=1")]
+    #[case::version_minor("application/vnd.miden; version=1.2")]
+    #[case::version_wildcard_major("application/vnd.miden; version=*")]
+    #[case::version_wildcard_minor("application/vnd.miden; version=1.*")]
+    #[case::version_wildcard_patch("application/vnd.miden; version=1.2.*")]
+    #[case::matching_network(
+        "application/vnd.miden; network=0x00000000000000000000000000000000000000000000000000000000deadbeef"
```

Comment:
Should this be `genesis=`?


```suggestion
        "application/vnd.miden; genesis=0x00000000000000000000000000000000000000000000000000000000deadbeef"
```

I guess it works anyway because it's not validated in any way, but not sure if that was the intention.